src/journal/journald-server.c

```diff
@@ -268,13 +268,14 @@ static int open_journal(
 }
 
 static int system_journal_open(Server *s, bool flush_requested) {
+        bool flushed = false;
         const char *fn;
         int r = 0;
 
         if (!s->system_journal &&
             (s->storage == STORAGE_PERSISTENT || s->storage == STORAGE_AUTO) &&
             (flush_requested
-             || access("/run/systemd/journal/flushed", F_OK) >= 0)) {
+             || (flushed = (access("/run/systemd/journal/flushed", F_OK) >= 0)))) {
```

Comment:
i figure this could be more readable if we'd hide this in a new tiny function flush_flag_set() or so, that hides the access() thing inside....